test/e2e/page-objects/pages/settings/contacts-settings.ts

```diff
@@ -24,6 +48,99 @@ class ContactsSettings {
     }
     console.log('Contacts Settings page is loaded');
   }
+
+  /**
+   * Adds a new contact to the address book.
+   *
+   * @param userName - The name of the contact.
+   * @param address - The address of the contact.
+   */
+  async addContact(userName: string, address: string): Promise<void> {
+    console.log('Adding new contact on contacts settings page');
+    await this.driver.clickElement(this.addContactButton);
+    await this.driver.fill(this.userNameInput, userName);
+    await this.driver.fill(this.createContactAddressInput, address);
+    await this.driver.clickElementAndWaitToDisappear(
+      this.confirmAddContactButton,
+    );
+  }
+
+  /**
+   * Deletes a contact from the address book.
+   *
+   * @param contactName - The name of the contact to delete.
+   */
+  async deleteContact(contactName: string): Promise<void> {
+    console.log('Deleting contact on contacts settings page');
+    await this.driver.clickElement({
+      text: contactName,
+      css: this.contactListItem,
+    });
+    await this.driver.clickElement(this.editContactButton);
+    await this.driver.clickElementAndWaitToDisappear(this.deleteContactButton);
+  }
+
+  /**
+   * Edits a contact in the address book.
+   *
+   * @param contactName - The name of the contact to edit.
+   * @param userName - The new name of the contact.
+   * @param address - The new address of the contact.
+   */
+  async editContact(
+    contactName: string,
+    userName: string,
+    address: string,
+  ): Promise<void> {
+    console.log('Editing contact on contacts settings page');
+    await this.driver.clickElement({
+      text: contactName,
+      css: this.contactListItem,
+    });
+    await this.driver.clickElement(this.editContactButton);
+    await this.driver.fill(this.userNameInput, userName);
+    await this.driver.fill(this.editContactAddressInput, address);
+    await this.driver.clickElementAndWaitToDisappear(
+      this.confirmAddContactButton,
+    );
+  }
+
+  /**
+   * Check if a contact is displayed on the contacts settings page.
+   *
+   * @param params - The parameters object
+   * @param params.userName - The name of the contact.
```

Comment:
super small nit: what about `contactName` here?

Reply:
make sense, addressed in the latest commit [344c48c](https://github.com/MetaMask/metamask-extension/pull/33022/commits/344c48c10b7dc5bc53d4625d28f1cfc13133856c)